### PR TITLE
feat(zip)!: add `includeSources` and rename `ignoredSources` to `excludeSources`

### LINKED ISSUE #377: Chunk hashes change when building from sources ZIP for Firefox Review
See #373

### PR DESCRIPTION
**Pull request by dvlden**
This PR resolves #377 

I am not sure is there a way to contact you somehow? Maybe Discord? 
Cause I'd like to learn if there's an easy way to test the source changes locally?

For this PR I did `pnpm run build` and then I linked from my test project the wxt source using `pnpm link ../wxt`

### CODE REVIEW

**Comment by aklinker1** [comment]
Hmm, I don't want to include `.env` files in zip... They usually contain secrets that shouldn't be shared. Is simply the presencse of a `.env` file enough to change the ZIP? Or are you using `import.meta.env` somewhere, and that's causing the ZIP to be different?

**Comment by aklinker1** [comment]
> I am not sure is there a way to contact you somehow? Maybe Discord?

Hmm, I have a discord, but I don't have notifications on, and I haven't opened it for 2 years. I don't like being overran by notifications. I'd prefer just starting a discussion on github, but maybe I should make a discord channel for WXT, it's pretty common for libraries to do that...

**Comment by aklinker1** [comment]
> I'd like to learn if there's an easy way to test the source changes locally?

The easiest way to test the app is by recreating the issue in the `demo/` folder, and testing there. Once it's working there, to test in a real project, I run `pnpm pack` to get a tarball, then I copy it into the project, and install the tarball.

**Comment by dvlden** [comment]
Hmmm, is there part of WXT that can be used for server-side parts? If not, I do not see a problem of an `.env` being zipped for review team, since it all ends up bundled in the code anyways.

Even if there was a scenario that someone would commit the .env file to the repo, if it's all client-side code, it's not considered a secret.

**Comment by aklinker1** [comment]
> Even if there was a scenario that someone would commit the .env file to the repo, if it's all client-side code, it's not considered a secret.

@dvlden I use `.env` files to store default login credentials for development, like a GitHub personal access token:

https://github.com/aklinker1/github-better-line-counts/blob/552a0db045fbf0159bb13e068938c65dc9feff74/src/utils/storage.ts#L22-L24

**Comment by dvlden** [comment]
Ouch, I see... In that case, your idea to have includes as an option sounds good to me.

Let me know if this is something that you'd let me try to implement? Cause it's similar to excludes, just opposite.

**Comment by aklinker1** [comment]
Yeah, if you want to give it a go, that'd be great!

**Comment by dvlden** [comment]
I did not come across any tests for this? I thought it may break, since I renamed `ignoredSources`, but no failing tests. 

Where do I update documentation? Is that inferred from DocBlock?

What do you think about it? I tested in `demo` dir and my project and it seems to work.

File: src/types/external.ts [hunk 1]
```diff
@@ -143,6 +143,19 @@ export interface InlineConfig {
      * @default config.root
      */
     sourcesRoot?: string;
+    /**
+     * [Minimatch](https://www.npmjs.com/package/minimatch) patterns of files to include when
+     * creating a ZIP of all your source code for Firfox. Patterns are relative to your
+     * `config.zip.sourcesRoot`.
+     *
+     * Many files/dirs are already excluded by default. Use this to override exclusion of files/directories that are excluded by default.
```
Comment (aklinker1) [inline src/types/external.ts:151]:
```suggestion
     * [Minimatch](https://www.npmjs.com/package/minimatch) patterns of files to include when
     * creating a ZIP of all your source code for Firefox. Patterns are relative to your
     * `config.zip.sourcesRoot`.
     *
     * Included sources override `excludedSources`. So if a file matches both lists, it is included in the ZIP.
```

File: src/core/utils/testing/fake-objects.ts [hunk 2]
```diff
@@ -232,7 +232,8 @@ export const fakeInternalConfig = fakeObjectCreator<InternalConfig>(() => {
     },
     zip: {
       artifactTemplate: '{{name}}-{{version}}.zip',
-      ignoredSources: [],
+      includedSources: [],
+      excludedSources: [],
```
Comment (aklinker1) [inline src/core/utils/testing/fake-objects.ts:236]: If we're making a breaking change, can we change this to `includeSources` and `excludeSources`? I think most config fields don't have the `d` at the end. Here's vitest:

Screenshot 2024-01-29 at 10 42 49 AM

**Review by aklinker1 — CHANGES_REQUESTED** [review]
Looks good, could of small changes. No, there are no tests for this right now, and yes, any JSDoc is automatically updated on the website.

File: src/core/utils/testing/fake-objects.ts (hunk 2 above)
Comment (dvlden) [reply src/core/utils/testing/fake-objects.ts:236]: Oh yes, definitely. I wanted to do this, but given it was `ignored` (with a `d`), I thought you may have preferred it that way. I'm on it.

**Review by aklinker1 — APPROVED** [review]
Nice, thanks 👍 I'll get this released soon.

**Comment by dvlden** [comment]
> Nice, thanks 👍 I'll get this released soon.

Glad to be of any help. Looking forward to more contributions. If you get something that you think I could resolve in this large project's codebase, feel free to assign me. 

Really need this to get through Firefox approvals.

**Pull request merged**